frontend/alert/components/AlertForm/index.tsx

```diff
@@ -148,25 +149,51 @@ const AlertForm = ({
         setSelectedCourses(new Set());
     };
 
+    /**
+     * Clear the input value and setValue
+     * @param newSelectedCourses - most up-to-date selected courses set
+     * @param suggestion - the section
+     */
+     const clearInputValue = () => {
+        if (autoCompleteInputRef.current) {
+            autoCompleteInputRef.current.value = "";
+            setValue("");
+        }
+    }
+
     const deselectCourse = (section: Section): boolean => {
         const newSelectedCourses = new Set(selectedCourses);
         const removed = newSelectedCourses.delete(section);
         removed && setSelectedCourses(newSelectedCourses);
+
+        if (newSelectedCourses.size === 0) {
+            clearInputValue();
+        }
+
         return removed;
     }
 
     const submitRegistration = () => {
         // if user has a auto fill section and didn't change the input value then register for section
+        // and support user manually entered a course (without checking checkbox)
         if (
             autoCompleteInputRef.current &&
-            autoCompleteInputRef.current.value === autofillSection
+            (autoCompleteInputRef.current.value === autofillSection || (autoCompleteInputRef.current.value !== "" && selectedCourses.size == 0))
         ) {
             doAPIRequest("/api/alert/registrations/", "POST", {
-                section: autofillSection,
+                section: autoCompleteInputRef.current.value,
                 auto_resubscribe: autoResub === "true",
             })
-                .then(setResponse)
+                .then((res) => {
+                    if (res.status === 200) {
+                        clearInputValue();
+                    }
```

Comment:
See below, this should probably be:
```
if (res.ok) { 
    clearInputValue();
}
```
The [Create Registration](https://penncoursealert.com/api/documentation/#operation/Create%20Registration) route returns 201 on success (not 200). The `ok` field of a `Response` object is true iff the response code is 2xx.